### PR TITLE
Add team unarchive and delete events

### PR DESCRIPTION
**Pull request by clearab**

### CODE REVIEW

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md [hunk 1]
```diff
@@ -38,7 +38,9 @@ The following table shows a list of Teams conversation update events, with links
 | members added   | membersAdded   | OnTeamsMembersAddedAsync   | [A member added](#team-members-added)   | All |
 | members removed | membersRemoved | OnTeamsMembersRemovedAsync | [A member was removed](#team-members-removed) | groupChat & team |
 | team renamed        | teamRenamed       | OnTeamsTeamRenamedAsync    | [A Team was renamed](#team-renamed)       | Team |
+| team deleted        | teamDeleted       | OnTeamsTeamDeletedAsync    | [A Team was deleted](#team-deleted)       | Team |
 | team archived        | teamArchived       | OnTeamsTeamArchivedAsync    | [A Team was archived](#team-archived)       | Team |
+| team unarchived        | teamUnarchived       | OnTeamsTeamUnarchivedAsync    | [A Team was unarchived](#team-unarchived)       | Team |
 | team restored        | teamRestored      | OnTeamsTeamRestoredAsync    | [A Team was renamed](#team-renamed)       | Team |
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:44]: The description here is "A team was renamed"

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md [hunk 2]
```diff
@@ -706,6 +708,80 @@ async def on_teams_team_renamed(
...
+        "id": "19:efa9296d959346209fea44151c742e73@thread.skype"
+    },
+    "recipient": { 
+        "id": "28:f5d48856-5b42-41a0-8c3a-c5f944b679b0",
+        "name": "SongsuggesterLocal"
+    },
+    "channelData": {
+        "team": {
+            "id": "19:efa9296d959346209fea44151c742e73@thread.skype",
+            "name": "Team Name"
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:763]: We've verified that team name is included, right? That's awesome, and necessary for delete scenarios.

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md [hunk 3]
```diff
@@ -784,6 +860,85 @@ async def on_teams_team_archived(
...
+	return await turn_context.send_activity(
+		MessageFactory.text(f"The team name is {team_info.name}")
+	)
+```
+
+* * *
+
 ### Team restored
 
 The bot receives a notification when the team it is installed in is restored. It receives a `conversationUpdate` event with `eventType.teamrestored` in the `channelData` object.
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:944]: Recommend changing to "... it is installed in is restored after having been deleted" (or similar, so it's clear that restore means undelete)

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md [hunk 4]
```diff
@@ -706,6 +708,80 @@ async def on_teams_team_renamed(
 
 * * *
 
+### Team deleted
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:711]: Should we move this to be next to Team restored, to emphasize that the relationship between the two?

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md (hunk 1 above)
Comment (clearab) [reply msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:44]: fixed

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md (hunk 2 above)
Comment (clearab) [reply msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:763]: Asking Tatiana/Yunny to validate the payloads.

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md (hunk 3 above)
Comment (clearab) [reply msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:944]: changed

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md (hunk 4 above)
Comment (clearab) [reply msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:711]: moved

**Review by aosolis — APPROVED** [review]

File: msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md (hunk 2 above)
Comment (aosolis) [reply msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md:763]: Rest look good to me. Thanks Andrew!

**Review by KirtiPereira — APPROVED** [review]
Approved.

**Review by surbhigupta12 — APPROVED** [review]
Approving the PR.

**Pull request merged**